packages/react/src/components/IconButton/index.tsx

```diff
@@ -1,52 +1,55 @@
-import React, { useRef } from 'react';
-import PropTypes from 'prop-types';
+import React, {
+  useRef,
+  forwardRef,
+  useImperativeHandle,
+  MutableRefObject
+} from 'react';
 import classnames from 'classnames';
 import { Placement } from '@popperjs/core';
 import Icon from '../Icon';
 import Tooltip from '../Tooltip';
-
 export interface IconButtonProps
   extends React.ButtonHTMLAttributes<HTMLButtonElement> {
   icon: string;
   label: string;
   tooltipPlacement?: Placement;
 }
 
-export default function IconButton({
-  icon,
-  label,
-  tooltipPlacement = 'auto',
-  className,
-  ...other
-}: IconButtonProps) {
-  const buttonRef = useRef<HTMLButtonElement>(null);
-  return (
-    <React.Fragment>
-      <button
-        type={'button'}
-        className={classnames('IconButton', className)}
-        ref={buttonRef}
-        {...other}
-      >
-        <Icon type={icon} />
-      </button>
-      <Tooltip
-        target={buttonRef}
-        placement={tooltipPlacement}
-        association="aria-labelledby"
-        hideElementOnHidden
-      >
-        {label}
-      </Tooltip>
-    </React.Fragment>
-  );
-}
+const IconButton = forwardRef<HTMLButtonElement, IconButtonProps>(
+  (
+    {
+      icon,
+      label,
+      tooltipPlacement = 'auto',
+      className,
+      ...other
+    }: IconButtonProps,
+    ref
+  ): JSX.Element => {
+    const buttonRef = useRef() as MutableRefObject<HTMLButtonElement>;
+    useImperativeHandle(ref, () => buttonRef.current);
 
+    return (
+      <React.Fragment>
+        <button
+          type={'button'}
+          className={classnames('IconButton', className)}
+          ref={buttonRef}
+          {...other}
+        >
+          <Icon type={icon} />
+        </button>
+        <Tooltip
+          target={buttonRef}
+          placement={tooltipPlacement}
+          association="aria-labelledby"
+          hideElementOnHidden
+        >
+          {label}
+        </Tooltip>
+      </React.Fragment>
+    );
+  }
+);
 IconButton.displayName = 'IconButton';
-
-IconButton.propTypes = {
-  icon: PropTypes.string,
-  label: PropTypes.string,
-  tooltipPlayemnt: PropTypes.string,
-  buttonRef: PropTypes.any
-};
```

Comment:
I don't care about `propTypes` here as I find them pretty useless, but others have complained when they were removed. cc @schne324

Reply:
prop types were being problematic with forwardRef + Typescript, and they become extraneous

Reply:
also: `tooltipPlayemnt`

Reply:
I can put them back, however they were not even properly enforcing required props.

Reply:
Put them back, the rest of the library has them so for now (even though I think they're pointless) let's keep them

Reply:
word. i'll fix them up to enforce the required props n stuff

Reply:
Added the PropTypes object back.